pkg/chains/formats/slsa/v2alpha2/internal/build_definitions/taskrun.go

```diff
@@ -0,0 +1,65 @@
+package builddefinitions
+
+import (
+	"fmt"
+
+	"github.com/tektoncd/chains/pkg/chains/objects"
+)
+
+type TaskBuildType struct {
+	BuildType          string
+	Tro                *objects.TaskRunObject
+	InternalParameters func(tro *objects.TaskRunObject) map[string]any
+}
+
+// internalParameters adds the tekton feature flags that were enabled
+// for the taskrun.
+func SLSATaskInternalParameters(tro *objects.TaskRunObject) map[string]any {
```

Comment:
Can we introduce something like `GetProvenance` to the TektonObject interface so we don't have to duplicate this logic for TaskRuns and PipelineRuns?

Reply:
Done!